docs/concepts/services-networking/dns-pod-service.md

```diff
@@ -262,157 +199,88 @@ spec:
-
-Linux's libc is impossibly stuck ([see this bug from
-2005](https://bugzilla.redhat.com/show_bug.cgi?id=168253)) with limits of just
-3 DNS `nameserver` records and 6 DNS `search` records.  Kubernetes needs to
-consume 1 `nameserver` record and 3 `search` records.  This means that if a
-local installation already uses 3 `nameserver`s or uses more than 3 `search`es,
-some of those settings will be lost.  As a partial workaround, the node can run
-`dnsmasq` which will provide more `nameserver` entries, but not more `search`
-entries.  You can also use kubelet's `--resolv-conf` flag.
-
-If you are using Alpine version 3.3 or earlier as your base image, DNS may not
-work properly owing to a known issue with Alpine. Check [here](https://github.com/kubernetes/kubernetes/issues/30215)
-for more information.
-
-## References
-
-- [Docs for the DNS cluster addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/dns/README.md)
-
-## What's next
-- [Autoscaling the DNS Service in a Cluster](/docs/tasks/administer-cluster/dns-horizontal-autoscaling/).
+### Pod's DNS Config
+
+Kubernetes v1.9 introduces an Alpha feature that allows users more control on
+the DNS settings for a Pod. To enable this feature, the cluster aministrator
+needs to enable the `CustomPodDNS` feature gate on the apiserver and the kubelet,
+for example, "`--feature-gates=CustomPodDNS=true,...`".
+When the feature gate is enabled, users can set the `dnsPolicy` field of a Pod
+to "`None`" and they can add a new field `dnsConfig` to a Pod Spec.
+
+The `dnsConfig` field is optional and it can work with any `dnsPolicy` settings.
+However, when a Pod's `dnsPolicy` is set to "`None`", the `dnsConfig` field has
+to be specified.
+
+Below are the properties a user can specify in the `dnsConfig` field:
+
+- `nameservers`: a list of IP addresses that will be used as DNS servers for the
+  Pod. There can be at most 3 IP addresses specified. When the Pod's `dnsPolicy`
+  is set to "`None`", the list must contain at least one IP address, otherwise
+  this property is optional.
+  The servers listed will be combined to the base nameservers generated from the
+  specified DNS policy with duplicate addresses removed.
+- `searches`: a list of DNS search domains for hostname lookup in the Pod.
+  This property is optional. When specified, the provided list will be merged
+  into the base search domain names generated from the chosen DNS policy.
+  Duplicate domain names are removed.
+  Kubernetes allows for at most 6 search domains.
+- `options`: an optional list of objects where each object may have a `name`
+  property (required) and a `value` property (optional). The contents in this
+  property will be merged to the options generated from the specified DNS policy.
+  Duplicate entries are removed.
+
+The following is an example Pod with custom DNS settings:
+
+{% include code.html language="yaml" file="ingress.yaml" ghlink="/docs/concepts/services-networking/custom-dns.yaml" %}
```

Comment:
s/ingress.yaml/custom-dns.yaml

BTW it might be better to give out the result resolv.conf content like https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/pod-resolv-conf.md#pod-api-examples does?